R/ci_validate_hub_config.R

```diff
@@ -0,0 +1,118 @@
+#' Validate Hub config files against hubverse schema
+#'
+#' A continuous integration helper
+#'
+#' @param hub_path path to the hub. Defaults to the value of the `HUB_PATH`
+#'   environment variable.
+#' @param gh_output path to a file that can record variables for use by other
+#'   actions. This defaults to the value of the `GITHUB_OUTPUT` environment
+#'   variable.
+#' @param ... arguments passed on to [validate_hub_config()].
+#'
+#' @return a logical value indicating if the hub config is valid or not. This
+#' has the side-effect of creating a file called `diff.md` in the `hub_path`
+#' directory. If running interactively, it will need to be cleaned up.
+#' @details
+#' This function is to be used within a continuous integration context.
+#' It is intended to be used in a workflow that checks the validity of
+#' a hub context. Below is an example setup of steps on GitHub Actions
+#' where the environment variables `PR_NUMBER` and `HUB_PATH` have been
+#' defined:
+#'
+#' ```yaml
+#'      - uses: actions/checkout@v4
+#'      - uses: r-lib/actions/setup-r@v2
+#'        with:
+#'          install-r: false
+#'          use-public-rspm: true
+#'          extra-repositories: 'https://hubverse-org.r-universe.dev'
+#'      - uses: r-lib/actions/setup-r-dependencies@v2
+#'        with:
+#'          cache: 'always'
+#'          packages: |
+#'            any::hubAdmin
+#'            any::sessioninfo
+#'      - name: Run validations
+#'        id: validate
+#'        run: hubAdmin::ci_validate_config()
+#'        shell: Rscript {0}
+#'      - name: "Comment on PR"
+#'        id: comment-diff
+#'        if: ${{ github.event_name != 'workflow_dispatch' }}
+#'        uses: carpentries/actions/comment-diff@main
+#'        with:
+#'          pr: ${{ env.PR_NUMBER }}
+#'          path: ${{ env.HUB_PATH }}/diff.md
+#'      - name: Error on Failure
+#'        if: ${{ steps.validate.outputs.result == 'false' }}
+#'        run: |
+#'          echo "::error title=Invalid Configuration::Errors were detected"
+#'          exit 1
+#' ```
```

Comment:
I think it would be nice to also point to the [`validate-config` hubverse action](https://github.com/hubverse-org/hubverse-actions/tree/main/validate-config)  and mention `hubCI::use_hub_github_action('validate-config')` as a way to add the workflow to a hub/package/repo.

Reply:
Done!